lib/link_checker.rb

```diff
@@ -0,0 +1,53 @@
+require 'yaml'
+require 'rest-client'
+require_relative 'root'
+require_relative 'app_config'
+require_relative 'url_checker'
+
+
+
+class LinkChecker
+  def initialize
+    puts "#{ROOT_DIR}/config/govuk_pages.yml"
+    @urls = YAML.load_file("#{ROOT_DIR}/config/govuk_pages.yml")
+    @config = AppConfig.new
+    @results = Hash.new { |hash, key| hash[key] = Array.new }
```

Comment:
This is not equivalent.

Hash.new([]) asssigns THE SAME empty Array to every new hash key, whereas the version in the code creates a new empty array for every new hash key.


irb(main):002:0* h = Hash.new([])
=> {}

irb(main):005:0* h[:a] << 'a1'
=> ["a1"]
irb(main):006:0> h[:a] << 'a2'
=> ["a1", "a2"]
irb(main):007:0> h[:b] << 'b'
=> ["a1", "a2", "b"]
irb(main):008:0>
irb(main):011:0>  h[:a]
=> ["a1", "a2", "b"]
irb(main):012:0> h[:b]
["a1", "a2", "b"]
=> ["a1", "a2", "b"]

as opposed to:
irb(main):047:0* h = Hash.new { |hash, key| hash[key] = Array.new }
=> {}
irb(main):048:0> h[:a] << 'a1'
=> ["a1"]
irb(main):049:0> h[:a] << 'a2'
=> ["a1", "a2"]
irb(main):050:0> h[:b] << 'b'
=> ["b"]
irb(main):051:0> h
=> {:a=>["a1", "a2"], :b=>["b"]}